qiskit_algorithms/gradients/utils.py

```diff
@@ -137,6 +137,13 @@ def _make_lin_comb_gradient_circuit(
                 lin_comb_circuit.h(qr_aux)
                 if add_measurement:
                     lin_comb_circuit.measure(qr_aux, cr_aux)
+                # This next line which assigns data is a workaround otherwise the
+                # circuit parameters may not be properly recognized as data is now
+                # managed in Rust and changing things may break parameters - making a
+                # copy of itself by assignment sorts things out.
+                # See https://github.com/Qiskit/qiskit/blob/main/releasenotes/notes
+                #     /circuit-gates-rust-5c6ab6c58f7fd2c9.yaml#L47-L79
+                lin_comb_circuit.data = lin_comb_circuit.data
```

Comment:
Not ideal but I agree that it's the most straightforward workaround so far.

Reply:
Ok I just talked to @Cryoris and I agree that the dagcircuit approach is nicer. Let's wait to see if his implementation makes it :)

Reply:
There is a PR against the branch in my fork to alter this for a DAG if you want to take a look https://github.com/woodsp-ibm/qiskit-algorithms/pull/1 If you and he feel its better with that as a replacement I am happy to have that instead. In the case of this PR I was thinking to leave it open until Qiskit 1.2 as I believed there was a possibility of that the problem may be alleviated. But either way perhaps the DAG approach is just better and we should go for that anyway.

Reply:
Using DAGs is slower unfortunately, so that's off the table for now 🙂